okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpWritableBufferAllocator.java

```diff
@@ -38,8 +38,24 @@
  * The default allocator for {@link OkHttpWritableBuffer}s used by the OkHttp transport.
  */
 class OkHttpWritableBufferAllocator implements WritableBufferAllocator {
+
+  private final int maxDataLength;
+
+  /**
+   * Construct a new instance
+   * @param maxDataLength the maximum allowed length of a DATA frame.
```

Comment:
Looking at this some more, it doesn't seem there will be any performance difference between the two. So I do think I would prefer us improving OutboundFlowState.writeBytes().